pkg/sst/sst.go

```diff
@@ -234,3 +238,124 @@ func getBits(val, i, j uint32) uint32 {
 func isBitSet(val, n uint32) bool {
 	return val&(1<<n) != 0
 }
+
+func setBit(val, n uint32) uint32 {
+	return val | (1 << n)
+}
+
+func clearBit(val, n uint32) uint32 {
+	return val &^ (1 << n)
+}
+
+func setBFStatus(info *SstPackageInfo, status bool) error {
+	rsp, err := sendMboxCmd(info.pkg.cpus[0], CONFIG_TDP, CONFIG_TDP_GET_TDP_CONTROL, uint32(info.PPCurrentLevel))
+	if err != nil {
+		return fmt.Errorf("failed to read SST status: %v", err)
+	}
+
+	req := clearBit(rsp, 17)
+	if status {
+		req = setBit(rsp, 17)
+	}
+
+	if _, err = sendMboxCmd(info.pkg.cpus[0], CONFIG_TDP, CONFIG_TDP_SET_TDP_CONTROL, req); err != nil {
+		return fmt.Errorf("failed to enable SST %s: %v", "BF", err)
+	}
+
+	info.BFEnabled = status
+
+	return nil
+}
+
+func setFreq2Max(info *SstPackageInfo) error {
+	for _, cpu := range info.BFCores.Members() {
+		err := setCPUMaxFreq(cpu)
+		if err != nil {
+			if os.IsNotExist(err) {
+				continue
+			} else {
+				sstlog.Debugf("sst-bf cpu %d set min freq error: %v", cpu, err)
+				return err
+			}
+		}
+
+		minScalingFreq, err := getCPUScalingMinFreq(cpu)
+		if err != nil {
+			sstlog.Debugf("sst-bf cpu %d set scaling freq error: %v", cpu, err)
+			return err
+		}
+
+		baseFreq, err := getCPUBaseFreq(cpu)
+		if err != nil {
+			sstlog.Debugf("sst-bf cpu %d set base freq error: %v", cpu, err)
+			return err
+		}
+
+		if minScalingFreq < baseFreq {
+			if err := setCPUScalingMinFreq(cpu, baseFreq); err != nil {
+				sstlog.Debugf("sst-bf cpu %d set min scaling freq error: %v", cpu, err)
+				return err
```

Comment:
Hmm, I'm now somehow missing the logic here 🤔  First (above) we set to max freq. After that, I wouldn't expect `minScalingFreq < baseFreq` to ever be true

Reply:
I clarified the function names etc, hopefully the code is more readable now.